### ISSUE TITLE
Add HistGradientBoostingRegressor

### ISSUE THREAD

**Issue by Thilakraj1998** [blobcity/autoai#61]
Add HistGradientBoostingRegressorinto the library.

Primary File to Change: https://github.com/blobcity/autoai/blob/main/blobcity/config/regressor_config.py

Reference HistGradientBoostingRegressor Implementation:
https://github.com/blobcity/ai-seed/blob/main/Regression/Histogram-Based%20Gradient%20Boosting%20Trees/HistGradientBoostingRegressor.ipynb

Official API Refer for parameters:
https://scikit-learn.org/stable/modules/generated/sklearn.ensemble.HistGradientBoostingRegressor.html#sklearn.ensemble.HistGradientBoostingRegressor

Dependencies if any, must be appropriately added. Test run of the train function on a regression problem must pass, and the function must attempt to train an HistGradientBoostingRegressor as a potential best fit model.
